src/NLog/Layouts/JsonLayout.cs

```diff
@@ -76,18 +76,29 @@ protected override string GetFormattedMessage(LogEventInfo logEvent)
             foreach (var col in this.Attributes)
             {
                 jsonWrapper.Inner = col.Layout;
+                jsonWrapper.JsonEncode = col.Encode;
                 string text = jsonWrapper.Render(logEvent);
-                
+
                 if (!string.IsNullOrEmpty(text))
                 {
-                    if (!first) 
+                    if (!first)
                     {
                         sb.Append(", ");
                     }
 
                     first = false;
 
-                    sb.AppendFormat("\"{0}\": \"{1}\"", col.Name, text);
+                    var format = "\"{0}\": \"{1}\"";
+
+                    //If encoding was disabled for current attribute,
+                    //Do not escape the value of the attribute.
+                    //This enables user to write already formated JSON.
```

Comment:
The really important difference is here that the escaped value is always a string? (the quotes).